optaplanner-wb-screens/optaplanner-wb-solver-editor/optaplanner-wb-solver-editor-backend/src/main/java/org/optaplanner/workbench/screens/solver/backend/server/SolverValidator.java

```diff
@@ -16,38 +16,108 @@
+    private LRUBuilderCache builderCache;
 
-    public List<ValidationMessage> validate( String xml ) {
-        ArrayList<ValidationMessage> validationMessages = new ArrayList<ValidationMessage>();
+    public SolverValidator() {
+    }
 
-        try {
-            SolverConfig solverConfig = configPersistence.toSolverConfig( xml );
-            solverConfig.buildSolver( new SolverConfigContext() );
+    @Inject
+    public SolverValidator( final @Named( "ioStrategy" ) IOService ioService,
+                            final KieProjectService projectService ) {
+        this.ioService = ioService;
+        this.projectService = projectService;
+    }
 
-        } catch (Exception e) {
-            validationMessages.add( make( e ) );
+    public List<ValidationMessage> validate( final Path resourcePath,
+                                             final String content ) {
+
+        // Run validation with the changed resource ( == everything builds ).
+        final KieProject kieProject = projectService.resolveProject( resourcePath );
+
+        final Validator validator = new Validator( new ValidatorFileSystemProvider( resourcePath,
+                                                                                    new ByteArrayInputStream( content != null ? content.getBytes( Charsets.UTF_8 ) : content.getBytes() ),
+                                                                                    kieProject,
+                                                                                    ioService,
+                                                                                    new JavaFileFilter() ) );
+
+        final List<ValidationMessage> validationMessages = validator.validate();
+
+        if ( validationMessages.isEmpty() ) {
+            final ValidationMessage validationMessage = createSolverFactory( resourcePath,
+                                                                             validator,
+                                                                             kieProject );
+            if ( validationMessage != null ) {
+                validationMessages.add( validationMessage );
+            }
         }
 
         return validationMessages;
     }
 
-    private ValidationMessage make( Exception e ) {
+    private ValidationMessage createSolverFactory( final Path resourcePath,
+                                                   final Validator validator,
+                                                   final KieProject kieWorkbenchProject ) {
+
+        final org.drools.compiler.kie.builder.impl.KieProject kieProject = new KieModuleKieProject( ( InternalKieModule ) validator.getKieBuilder().getKieModule(), null );
+        final KieContainer kieContainer = new KieContainerImpl( kieProject,
+                                                                KieServices.Factory.get().getRepository() );
+
+        try {
+
+            SolverFactory.createFromKieContainerXmlResource( kieContainer,
```

Comment:
At this point, doing

     kieContainer.getClassLoader().getPackage("org.projectA")

return null. so I don't think the kieContainer is build correctly. Furthermore, doing 

     kieContainer.getClassLoader().getPackages()

reveals it includes drools-compiler, mockito, aether and 900 other packages, so it clearly leaks the Test's classpath into the kieContainer's classpath (so in any case, this makes the test not representable).